src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -413,6 +413,51 @@ try
 }
 CATCH_RETURN()
 
+[[nodiscard]] HRESULT AtlasEngine::PaintSelections(const std::vector<til::rect>& rects) noexcept
+try
+{
+    // Unfortunately there's no step after Renderer::_PaintBufferOutput that
+    // would inform us that it's done with the last AtlasEngine::PaintBufferLine.
+    // As such we got to call _flushBufferLine() here just to be sure.
+    _flushBufferLine();
+
+    for (auto rect : rects)
+    {
+        const auto y = gsl::narrow_cast<u16>(clamp<til::CoordType>(rect.top, 0, _p.s->viewportCellCount.y));
+        const auto from = gsl::narrow_cast<u16>(clamp<til::CoordType>(rect.left, 0, _p.s->viewportCellCount.x - 1));
+        const auto to = gsl::narrow_cast<u16>(clamp<til::CoordType>(rect.right, from, _p.s->viewportCellCount.x));
+
+        if (y < _p.rows.size() && rect.top >= 0)
+        {
+            auto& row = *_p.rows[y];
+
+            auto it = std::find_if(row.searchSelections.begin(), row.searchSelections.end(), [&from](const SearchSelection& selection) {
+                return selection.from == from;
+            });
+
+            if (it != row.searchSelections.end())
+            {
+                if (to > it->to)
+                {
+                    it->to = to;
+                }
+            }
+            else
+            {
+                auto s = SearchSelection{ from, to };
+                row.searchSelections.emplace_back(s);
+            }

```

Comment:
I think this might have been an oversight on my end.  Previously I was having trouble clearing out the selections from prior searches and this was left over from that.  I commented out the code and tried a bunch of different searches and didn't notice any overlaps.

Reply:
actually it looks like I still have an issue with selections not being cleared.  With the code commented,  when I escape out of the search the highlight keeps getting darker each time render is called.

Reply:
I think this might be a decent alternative to what I had.

https://github.com/microsoft/terminal/pull/16227/commits/4cbd3c19ba35343bbf6b298fadbf5625d33720d1

Reply:
I think this is the last update,  when I had over 10,000 rows and searched for e,  the scrolling was somewhat choppy.  Updating GetSelectionRects to only include the rectangles that are in the visible viewport seems to fix the choppiness.

https://github.com/microsoft/terminal/pull/16227/commits/8bce03b6c3caec5630acf80f36865441fec0ae54